pkg/docker/config/config.go

```diff
@@ -83,6 +84,24 @@ func (d *DockerConfigEntry) UnmarshalJSON(data []byte) error {
 	return err
 }
 
+func encodeAuthField(d DockerConfigEntry) string {
+	basicAuth := fmt.Sprintf("%s:%s", d.Username, d.Password)
+	return base64.StdEncoding.EncodeToString([]byte(basicAuth))
+}
+
+// MarshalJSON marshals the entry with the basic auth field derived
+// from the username and password.
+func (d DockerConfigEntry) MarshalJSON() ([]byte, error) {
+	var tmp DockerConfigEntryWithAuth
+
+	tmp.Username = d.Username
+	tmp.Password = d.Password
+	tmp.Email = d.Email
+	tmp.Auth = encodeAuthField(d)
+
+	return json.Marshal(tmp)
+}
```

Comment:
Added this because we change the OCP global pull secret in the new tests, we need to read the current secret, modify it, and the apply it back to the cluster. The Auth field was missing before which was breaking the secret / OCP was not able to use the secret.